### PR TITLE
TRACING-3802| Add target allocator to opentelemetry

### PR DESCRIPTION
**Pull request by pavolloffay**
Version(s):

4.12, 4.13, 4.14, 4.15, 4.16

Issue:

https://issues.redhat.com/browse/TRACING-3802

Link to docs preview:

https://71074--ocpdocs-pr.netlify.app/openshift-enterprise/latest/otel/otel-configuring#otel-collector-ta-otel-configuring

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Review by IshwarKanse — APPROVED** [review]

File: modules/otel-config-target-allocator.adoc [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+// * otel/otel-configuring.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="otel-collector-ta-{context}"]
+= Target allocator
+
+The target allocator is an optional component of the OpenTelemetry Operator that shards scrape targets across the
+deployed fleet of collectors and integrates with the Prometheus `PodMonitor` and `ServiceMonitor` custom resources.
+When the target allocator is enabled,  the OpenTelemetry Operator adds `http_sd_config` to the enabled `prometheus` receiver that
+connects to the target allocator service.
```
Comment (max-cx) [inline modules/otel-config-target-allocator.adoc:12]:
```suggestion
The target allocator is an optional component of the OpenTelemetry Operator that shards scrape targets across the deployed fleet of OpenTelemetry Collector instances. The target allocatorintegrates with the Prometheus `PodMonitor` and `ServiceMonitor` custom resources. When the target allocator is enabled,  the OpenTelemetry Operator adds `http_sd_config` to the enabled `prometheus` receiver that connects to the target allocator service.
```

**Review by max-cx — APPROVED** [review]

**Comment by kalexand-rh** [comment]
4.11 is now [EOL](https://access.redhat.com/support/policy/updates/openshift), and we are no longer updating it. Please set the versions to 4.12+.

**Comment by max-cx** [comment]
I have incorporated the new file `modules/otel-config-target-allocator.adoc` from this PR into https://github.com/openshift/openshift-docs/pull/71565.

**Pull request closed without merging**